### PR TITLE
enhance feature extractor

### PR DESCRIPTION
**Pull request by BY571**
adding layer norm and dropout as options to MLP and CNN feature extractor.

### CODE REVIEW

File: pytorchrl/agent/actors/feature_extractors/cnn.py [hunk 1]
```diff
@@ -34,6 +34,8 @@ def __init__(self,
                  rgb_norm=True,
                  activation=nn.ReLU,
                  final_activation=True,
+                 dropout=0.0,
```
Comment (albertbou92) [inline pytorchrl/agent/actors/feature_extractors/cnn.py:37]: add parameter description in the docstrings

File: pytorchrl/agent/actors/feature_extractors/cnn.py [hunk 2]
```diff
@@ -34,6 +34,8 @@ def __init__(self,
...
+                 layer_norm=False,
```
Comment (albertbou92) [inline pytorchrl/agent/actors/feature_extractors/cnn.py:38]: add parameter description in the docstrings

File: pytorchrl/agent/actors/feature_extractors/mlp.py [hunk 3]
```diff
@@ -34,7 +34,12 @@ def __init__(self, input_space, hidden_sizes=[256, 256], output_size=256, activa
         layers = []
         sizes = [np.prod(input_shape)] + hidden_sizes + [output_size]
         for j in range(len(sizes) - 1):
-            layers += [nn.Linear(sizes[j], sizes[j + 1])]
+            layers += [nn.Linear(sizes[j], sizes[j + 1])
+                       ]
```
Comment (albertbou92) [inline pytorchrl/agent/actors/feature_extractors/mlp.py:38]: move ] to line 37

File: pytorchrl/agent/actors/off_policy_actor.py [hunk 4]
```diff
@@ -460,16 +460,18 @@ def create_critic(self, name):
         name : str
             Critic network name.
         """
-
+        if self.obs_feature_extractor:
+            obs_feature_extractor = self.obs_feature_extractor
+        else:
+            obs_feature_extractor =  default_feature_extractor(self.input_space)
```
Comment (albertbou92) [inline pytorchrl/agent/actors/off_policy_actor.py:466]: add blank space below line 466

File: pytorchrl/agent/actors/off_policy_actor.py [hunk 5]
```diff
@@ -535,25 +538,25 @@ def create_policy(self, name):
...
-
-        policy_common_feature_extractor = self.common_feature_extractor(
-            feature_size, **self.common_feature_extractor_kwargs)
-
+        if self.common_feature_extractor:
+            policy_common_feature_extractor = self.common_feature_extractor(
+                feature_size, **self.common_feature_extractor_kwargs)
+            feature_size = policy_common_feature_extractor(torch.zeros(1, feature_size)).shape[-1]
+        else:
+            policy_common_feature_extractor = nn.Identity()
```
Comment (albertbou92) [inline pytorchrl/agent/actors/off_policy_actor.py:557]: add blank space below line 557

**Review by albertbou92 — CHANGES_REQUESTED** [review]
Minimal changes and docstrings

**Review by albertbou92 — APPROVED** [review]
